Use the latest models from OpenAI

vidluther opened the pull request:
## 🧢 Changes

* add o3-mini
* add o1-mini 
* remove GPT 3-5 Turbo
* remove GPT Turbo
* remove GPT 4o 

Update the service tests to make sure they pass.

vidluther committed:
update(ai): show latest openAI models and remove outdated models

Replace outdated OpenAI model options with new recommendations and 
simplify naming. Update the display labels for models to improve 
clarity and align with current offerings. Remove obsolete options 
to streamline the selection process.

vidluther committed:
Merge pull request #7 from vidluther/update-openai-models

update(ai): show latest openAI models and remove outdated models

vidluther committed:
Merge branch 'gitbutlerapp:master' into master

vidluther committed:
feat: Update AI model option labels

Adjust the labels for OpenAI model options in the settings page to
use lowercase letters for "o3" and "o1" in "o3 Mini" and "o1 Mini"
respectively. This change ensures consistency with official model
naming conventions.

vidluther committed:
Merge pull request #8 from vidluther/update-openai-model-label-capitalization

feat: Update AI model option labels

vidluther committed:
feat: update default OpenAI model to O3mini

Updates the default OpenAI model from GPT-3.5-Turbo to O3mini in
both the AIService and the default GitConfig for tests. This change
aligns the application with the latest preferred model for improved
performance and efficiency.

vidluther committed:
Merge pull request #9 from vidluther/update-openai-model-to-o3mini

feat: update default OpenAI model to O3mini

estib-vega commented on apps/desktop/src/lib/ai/service.test.ts:
```diff
@@ -34,7 +34,7 @@ import type { SecretsService } from '$lib/secrets/secretsService';
 	[GitAIConfigKey.ModelProvider]: ModelKind.OpenAI,
 	[GitAIConfigKey.OpenAIKeyOption]: KeyOption.ButlerAPI,
-	[GitAIConfigKey.OpenAIModelName]: OpenAIModelName.GPT35Turbo,
+	[GitAIConfigKey.OpenAIModelName]: OpenAIModelName.O3mini,
```
Let's set 4o mini as a default instead, please

vidluther replied:
@estib-vega I just remembered the cost of o3-mini and o1-mini is the same.. they even recommend using o3-mini on the o1-mini page.. https://platform.openai.com/docs/models/o1-mini

"The o1 reasoning model is designed to solve hard problems across domains. o1-mini is a faster and more affordable reasoning model, but we recommend using the newer o3-mini model that features higher intelligence at the same latency and price as o1-mini." 

I can set it to o1-mini if you prefer.. but just wanted to point this out first.

estib-vega replied:
@vidluther I was talking about **4o mini** :) That one is waaay cheaper

vidluther replied:
oh yea.. duh

vidluther replied:
The changes are now in my master branch, so I think this PR gets updated automatically now? I'm not 100% sure how this GH/PR magic works..

estib-vega commented on apps/desktop/src/lib/ai/service.ts:
```diff
@@ -123,7 +123,7 @@ export class AIService {
 		return await this.gitConfig.getWithDefault<OpenAIModelName>(
 			GitAIConfigKey.OpenAIModelName,
-			OpenAIModelName.GPT35Turbo
+			OpenAIModelName.O3mini
```
Same here

vidluther committed:
Merge branch 'gitbutlerapp:master' into master

vidluther committed:
feat: change default AI model to GPT 4o mini

Updates the default OpenAI model from o3 Mini to GPT 4o mini and
marks it as the recommended option in the UI. This change reflects
our current recommendation for the best balance of performance and
capabilities for users.

vidluther committed:
Merge pull request #10 from vidluther/update-openai-recommended-model-to-gpt4o-mini

feat: change default AI model to GPT 4o mini

vidluther committed:
Merge branch 'master' into master

estib-vega commented:
@vidluther Nice!
Thanks a lot!
